Source/GSXib5KeyedUnarchiver.m

```diff
@@ -2917,6 +2943,79 @@ - (id) decodeTransitionStyle: (GSXibElement *)element
   return num;  
 }
 
+- (id) _decodePlacementForObject: (id)obj
+{
+  NSNumber *num = [NSNumber numberWithInteger: 0];
+  if ([obj isEqualToString: @"inherited"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementInherited];
+    }
+  else if ([obj isEqualToString: @"leading"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementLeading];
+    }
+  else if ([obj isEqualToString: @"top"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementTop];
+    }
+  else if ([obj isEqualToString: @"trailing"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementTrailing];
+    }
+  else if ([obj isEqualToString: @"bottom"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementBottom];
+    }
+  else if ([obj isEqualToString: @"center"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementCenter];
+    }
+  else if ([obj isEqualToString: @"fill"])
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementFill];
+    }  
+  else // if not specified then assume none...
+    {
+      num = [NSNumber numberWithInteger: NSGridCellPlacementNone];
+    }
+  return num;
+}
+
+- (id) decodeXPlacementForElement: (GSXibElement *)element
+{
+  id obj = [element attributeForKey: @"xPlacement"];
+  return [self _decodePlacementForObject: obj];
+}
+
+- (id) decodeYPlacementForElement: (GSXibElement *)element
+{
+  id obj = [element attributeForKey: @"yPlacement"];
+  return [self _decodePlacementForObject: obj];
+}
+
+- (id) decodeRowAlignmentForElement: (GSXibElement *)element
+{
+  id obj = [element attributeForKey: @"rowAlignment"];
+  NSNumber *num = [NSNumber numberWithInteger: 0];
```

Comment:
Why not something like this:

NSGridRowAlignment align;
if ([obj isEqualToString: @"inherited"])
     {
       align = NSGridRowAlignmentInherited;
     }
   else if ([obj isEqualToString: @"firstBaseline"])
     {
       align = NSGridRowAlignmentFirstBaseline;
     }
   else if ([obj isEqualToString: @"lastBaseline"])
     {
       align = NSGridRowAlignmentLastBaseline;
     }
   else
     {
       align = NSGridRowAlignmentNone;
     }
   return [NSNumber numberWithInteger: align];

Reply:
Yep, doing that now.

Reply:
Fixed there and one other place where I was doing the same thing.

Reply:
Pushed